### PR TITLE
Add benchmark for appending transactions

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**

### CODE REVIEW

File: writeaheadlog_test.go [hunk 1]
```diff
@@ -1018,7 +1026,47 @@ func BenchmarkTransactionSpeed100(b *testing.B) {
...
+}
+
+// BenchmarkTransactionSpeedAppend1000 runs benchmarkTransactionSpeed with 1000
+// and calls txn.Append
+//
+// Results (Model, txn/s, maxLatency(ms), date)
+//
+//
+func BenchmarkTransactionSpeedAppend1000(b *testing.B) {
+	benchmarkTransactionSpeed(b, 1000, true)
```
Comment (lukechampine) [inline writeaheadlog_test.go]: these should use sub-benchmarks:
```go
func BenchmarkTransactionSpeedAppend(b *testing.B) {
    numCalls := []int{1, 10, 100, 1000}
    for _, n := range numCalls {
        b.Run(strconv.Itoa(n), func(b *testing.B) {
            benchmarkTransactionSpeed(b, n, true)
        })
    }
}
```

**Comment by lukechampine** [comment]
I gotta say, I don't really like the way we're benchmarking things. Go's benchmarking tooling is most useful when you're benchmarking a single operation in isolation, because that enables you to identify which operations might be bottlenecks. In contrast, most of our benchmarks are running a bunch of operations in tandem. The fact that our benchmarks don't even use `b.N` should indicate that we aren't using the tooling for its intended purpose. They are still useful, because they give us a general sense of how fast the WAL is under certain conditions; and as we optimize things, we check the numbers again and hope that they go down. But we could accomplish the same thing by looking at the execution time of `TestSilo` (or some other general-purpose test). So I suggest that we shift our benchmarking focus to individual operations (like `Append`), and lean on `TestSilo` and other long-running tests to get a sense of the WAL's real-world performance.

**Comment by lukechampine** [comment]
The bolt project has an interesting approach to this: https://github.com/boltdb/bolt/blob/9145e047707a32d3685f15d9fd97e1b25445ba1b/cmd/bolt/main.go#L900

Basically, instead of an automated benchmark, bolt provides a standalone executable that runs a synthetic benchmark suite. I see two advantages of this: you can write an exhaustive benchmark without cluttering/slowing your automated tests, and you make it easy for anyone to run the benchmark on their system, since the binary can be provided directly instead of the user having to install the Go toolchain.

**Comment by ChrisSchinnerl** [comment]
That's a good point. Maybe the threads in the sub benchmarks should not iterate until 1 minute passed but for `b.N` iterations instead?

**Comment by DavidVorick** [comment]
I would be in support of converting our existing benchmark functions into a binary that we can use to run a benchmark suite. That would make it a lot easier to test on various drives and computers too.

**Pull request merged**